core/types/transaction_kroma.go

```diff
@@ -0,0 +1,46 @@
+package types
+
+import (
+	"math/big"
+)
+
+func (tx *Transaction) ToKromaDepositTx() *Transaction {
+	var kromaDep *KromaDepositTx
+	if dep, ok := tx.inner.(*DepositTx); ok {
+		kromaDep = &KromaDepositTx{
+			SourceHash: dep.SourceHash,
+			From:       dep.From,
+			To:         dep.To,
+			Mint:       nil,
+			Value:      new(big.Int),
+			Gas:        dep.Gas,
+			Data:       dep.Data,
+		}
+		if dep.Mint != nil {
+			kromaDep.Mint = new(big.Int).Set(dep.Mint)
+		}
+		if dep.Value != nil {
+			kromaDep.Value.Set(dep.Value)
+		}
+	} else if dep, ok := tx.inner.(*depositTxWithNonce); ok {
+		kromaDep = &KromaDepositTx{
+			SourceHash: dep.SourceHash,
+			From:       dep.From,
+			To:         dep.To,
+			Mint:       nil,
+			Value:      new(big.Int),
+			Gas:        dep.Gas,
+			Data:       dep.Data,
+		}
+		if dep.Mint != nil {
+			kromaDep.Mint = new(big.Int).Set(dep.Mint)
+		}
+		if dep.Value != nil {
+			kromaDep.Value.Set(dep.Value)
+		}
+	} else {
+		return tx
```

Comment:
In this case, the functionality is not worked at all, so returning nil and error is more appropriate IMO, wdyt?

Reply:
This function will only be used for DepositTx, and the intention is to directly return the transaction if it is of the KromaDepositTx type.
I added error handling logic to check whether the transaction type is DepositTx or not.